time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java

```diff
@@ -190,42 +237,66 @@ private TDoubleArrayList createDoubleValues() {
                 if (Doubles.tryParse(token) != null) {
-                    dataTypes[i - 2] = TimeSeriesDataType.DOUBLE;
+                    dataTypes[i - fixedColumns] = TimeSeriesDataType.DOUBLE;
                     TDoubleArrayList doubleValues = createDoubleValues();
                     doubleValues.add(parseDouble(token));
-                    values[i - 2] = doubleValues;
+                    values[i - fixedColumns] = doubleValues;
                 } else {
-                    dataTypes[i - 2] = TimeSeriesDataType.STRING;
+                    dataTypes[i - fixedColumns] = TimeSeriesDataType.STRING;
                     List<String> stringValues = createStringValues();
                     stringValues.add(checkString(token));
-                    values[i - 2] = stringValues;
+                    values[i - fixedColumns] = stringValues;
                 }
             } else {
-                if (dataTypes[i - 2] == TimeSeriesDataType.DOUBLE) {
-                    ((TDoubleArrayList) values[i - 2]).add(parseDouble(token));
-                } else if (dataTypes[i - 2] == TimeSeriesDataType.STRING) {
-                    ((List<String>) values[i - 2]).add(checkString(token));
+                if (dataTypes[i - fixedColumns] == TimeSeriesDataType.DOUBLE) {
+                    ((TDoubleArrayList) values[i - fixedColumns]).add(parseDouble(token));
+                } else if (dataTypes[i - fixedColumns] == TimeSeriesDataType.STRING) {
+                    ((List<String>) values[i - fixedColumns]).add(checkString(token));
                 } else {
-                    throw assertDataType(dataTypes[i - 2]);
+                    throw assertDataType(dataTypes[i - fixedColumns]);
                 }
             }
         }
 
         void parseLine(List<String> tokens) {
-            for (int i = 2; i < tokens.size(); i++) {
+            for (int i = fixedColumns; i < tokens.size(); i++) {
                 String token = tokens.get(i) != null ? tokens.get(i).trim() : "";
                 parseToken(i, token);
             }
-            // empty last cell case
-            if (tokens.size() == names.size() + 1) {
-                parseToken(tokens.size(), "");
-            }
 
-            times.add(ZonedDateTime.parse(tokens.get(0)));
+            parseTokenTime(tokens);
+        }
+
+        void parseTokenTime(List<String> tokens) {
+            switch (csvParserConfig.timeFormat()) {
+                case DATE_TIME:
+                    times.add(ZonedDateTime.parse(tokens.get(0)).toInstant().toEpochMilli());
+                    break;
+                case FRACTIONS_OF_SECOND:
+                    Double time = Double.parseDouble(tokens.get(0)) * 1000;
+                    times.add(time.longValue());
+                    break;
+                case MILLIS:
+                    Double millis = Double.parseDouble(tokens.get(0));
+                    times.add(millis.longValue());
+                    break;
+            }
```

Comment:
Add a default case that throws an AssertionError with the unexpected timeFormat

Reply:
Done